website/content/contribute/module/index.md

```diff
@@ -0,0 +1,199 @@
+
+```
+flows: # Indicate the data flow(s) in which the control plane should consider using this module 
+- read  # optional
+- write # optional
+- copy  # optional
+```
+
+### capabilities
+
+`capabilites.credentials-managed-by` may be either `secret-provider` or `data-mesh-auto`.  Currently only `secret-provider` is supported, meaning that the module will receive a link to the data set credentials and must use the Secret Provider API to obtain the credentials. 
+TODO: Add API link
+TODO: /Users/sima/Work/Dev/go/sima-m4d/the-mesh-for-data/manager/apis/app/v1alpha1/m4dmodule_types.go has `automatic` andd not `data-mesh-auto`
+
+`capabilites.supportedInterfaces` lists the supported data services from which the module can read data and to which it can write 
+* `flow` field can be `read`, `write` or `copy`
+* `protocol` field can take a value such as `kafka`, `s3`, `jdbc-db2`, `m4d-arrow-flight`, etc.
+* `format` field can take a value such as `avro`, `parquet`, `json`, or `csv`.
+Note that a module that targets copy flows will omit the `api` field and contain just `source` and `sink`, a module that only supports reading data assets will omit the `sink` field and only contain `api` and `source`
+
+`capabilites.api` indicates the protocol and data format supported for reading or writing data from the user's workload.
+
+An example for a module that copies data from a db2 database table to an s3 bucket in parquet format.
+```
+capabilities:
+    credentials-managed-by: secret-provider
+    supportedInterfaces:
+    - flow: copy  
+      source:
+        protocol: jdbc-db2
+        dataformat: table
+      sink:
+        protocol: s3
+        dataformat: parquet
+```
+
+An example for a module that has an API for reading data, and supports reading both parquet and csv formats from s3.
+```
+capabilities:
+    credentials-managed-by: secret-provider
+    api:
+      protocol: m4d-arrow-flight
+      dataformat: arrow
+    supportedInterfaces:
+    - flow: read
+      source:
+        protocol: s3
+        dataformat: parquet
+    - flow: read
+      source:
+        protocol: s3
+        dataformat: csv
+```
+
+`capabilites.actions`  are taken from a defined [Enforcement Actions Taxonomy](about:blank) 
+a module that does not perform any transformation on the data may omit the `capabilities.actions` field.
+
+The following is an example of how a module would declare that it knows how to redact, remove or encrypt data.  For each action there is a level indication, which can be data set level, column level, or row level.  In the example shown column level is indicated, and the actions arguments indicate the columns on which the transformation should be performed.
+
+```
+capabilities:
+    actions:
+    - id: "redact-ID"
+      name: "redact"
+      level: 2 # column
```

Comment:
We don't need both id and name, just id is enough.

Reply:
OK.  Removed names.  Please note though that they are still in the module yamls in the e2e test